### ISSUE TITLE
Why stop event propagation?

### ISSUE THREAD

**Issue by oliverjam** [fac25/week2-agencyProject-Patrick-Sumithra#18]
https://github.com/fac25/week2-agencyProject-Patrick-Sumithra/blob/b420506ab9c4d70a3260ffd79bd65db334226ba2/scripts/Contact/Modal.js#L14-L16

This isn't a criticism, just a question 😅 

Why exactly do you need to call `event.stopPropagation()` in the modal?

Sidenote: "why" explanations are a great thing to put in comments (instead of "what" comments). E.g. I can see _what_ this code does, but I don't know _why_ you need to do it

**Comment by patdel0** [comment]
Hi @oliverjam, 

To hide the modal whenever the user clicked outside of it, I had applied an event listener to the body element. 
Basically, whenever the body was clicked, the modal would be hidden from the user's view.

This solution resulted in the body click event being triggered when the modal itself was clicked. To prevent this from happening a 'hacky' fix was used, through the use of `event.stopPropagation()`.

Thanks to your issue making me revisit the code, I managed to come up with what I deem a better solution:
Create a guard clause within the body event listener function, that checks if any element related to the modal was clicked 021f298575d75c19afced7940430bd6abbc77b3d
